token/options.go

```diff
@@ -157,3 +160,32 @@ func WithKid(s string) Options {
 		return nil
 	}
 }
+
+// WithX5CFile returns a Options that sets the header x5c claims.
+func WithX5CFile(certFile string, key interface{}) Options {
+	return func(c *Claims) error {
+		if certFile == "" {
+			return errors.New("x5c-cert cannot be empty")
+		}
+		certs, err := pemutil.ReadCertificateBundle(certFile)
+		if err != nil {
+			return errors.Wrap(err, "error reading cert file for x5c header")
+		}
+
+		if err = x509util.VerifyCertKey(certs[0], key); err != nil {
+			return errors.Wrap(err, "x5c")
+		}
+
+		if certs[0].KeyUsage&x509.KeyUsageDigitalSignature == 0 {
+			return errors.New("x5c: certificate/private-key pair used to sign " +
+				"x5c token cannot be used for digital signature")
+		}
```

Comment:
For posterity (since we already had a similar discussion in certificates), Verify cannot check KeyUsage extensions (only ExtKeyUsage). I will however change this to check every certificate in the chain.

Reply:
Check that, I think only the leaf certificate should require the digital signature extension.